tests/unit/test_utilities.py

```diff
@@ -250,6 +250,21 @@ def test_value_case_matters(self):
         with pytest.raises(exceptions.KeyMismatchError):
             check_keys_match_recursive(a, b, [])
 
+    def test_match_list_items(self):
```

Comment:
Can you test for cases where non-strict matching should fail? perhaps `a=[1,2,3], b=["a"]`